File: src/agent/npc.py
"""
Shared nodes and types for NPCs.
"""
from __future__ import annotations

from typing import Literal, List

from pydantic import BaseModel


class NPCResponse(BaseModel):
    message: str
    emotion: Literal["relaxed", "nervous", "panicking", "angry", "upset", "depressed", "defensive"]
    topic: str
    summary: str
    knowledge: List[str] | None = None

File: src/agent/npc.ts
import { z } from "zod";

export const NPCEmotion = z.enum([
  "relaxed",
  "nervous",
  "panicking",
  "angry",
  "upset",
  "depressed",
  "defensive",
]);
export type NPCEmotion = z.infer<typeof NPCEmotion>;

export const NPCResponseSchema = z.object({
  message: z.string(),
  emotion: NPCEmotion,
  topic: z.string(),
  summary: z.string(),
  knowledge: z.array(z.string()).nullable().default(null),
});

export type NPCResponse = z.infer<typeof NPCResponseSchema>;
